data/campaigns/Eastern_Invasion/scenarios/12_Evacuation.cfg

```diff
@@ -172,186 +270,626 @@
+                    canrecruit=yes
+                    message= _ "We need help over here! Where are the rest of those lousy drakes? They should be in position by now!"
+                [/message]
+            [/then]
+        [/if]
+        {RESET_SIDE_AI 3 offensive 0.99 0.01} # remove the avoid
+
+        # only spawn drakes that have a wing-flapping standing_anim. Gliders count, but only when in the water
+        # and spawn a whole lot of them, so the player doesn't think "really? I’m supposed to be defeated by *that*?"
+        {MOVE_UNIT x,y=12,2 4 7}
+        {MOVE_UNIT x,y=13,1 5 5}
+        {MOVE_UNIT x,y=15,2 4 5}
+
+        {GENERIC_UNIT 3 (Drake Glider)     10 1} {ANIMATE}
+        {GENERIC_UNIT 3 (Sky Drake)         9 1}
+        {GENERIC_UNIT 3 (Hurricane Drake)   7 1} {ANIMATE}
+        {GENERIC_UNIT 3 (Sky Drake)         5 1}
+        {GENERIC_UNIT 3 (Hurricane Drake)   5 1} {ANIMATE}
+        {GENERIC_UNIT 3 (Sky Drake)         5 1}
+        {GENERIC_UNIT 3 (Hurricane Drake)   3 1} {ANIMATE}
+        {GENERIC_UNIT 3 (Sky Drake)         3 1}
+        {GENERIC_UNIT 3 (Sky Drake)         1 1} {ANIMATE}
+        {GENERIC_UNIT 3 (Hurricane Drake)   1 1}
+
+        {GENERIC_UNIT 3 (Drake Glider)      9 5} {ANIMATE}
+        {GENERIC_UNIT 3 (Drake Glider)      8 7}
+        {GENERIC_UNIT 3 (Drake Glider)      7 5} {ANIMATE}
+        {GENERIC_UNIT 3 (Hurricane Drake)   1 5}
+        {GENERIC_UNIT 3 (Sky Drake)         1 7} {ANIMATE}
+        {NAMED_UNIT   3 (Sky Drake)         2 7 "bridgeburner" "Gel’ka" ()}
+        {GENERIC_UNIT 3 (Sky Drake)         1 8} {ANIMATE}
+
         [message]
             speaker=Dacyn
-            message= _ "We have not thrown down the bridge in time! The orcs will kill us all!"
+            message= _ "We have not crossed the bridge in time. Now we will all be killed!"
         [/message]
+
+        # don't actually end in defeat here; give them a chance to move and attack
+        # they'll probably kill player units, but that's ok because the "turns over" event triggers immediately after
     [/event]
 
     [event]
         name=enemies defeated
         [message]
             speaker=Gweddry
-            message= _ "Well, we didn’t have to blow up the bridge after all. Now we must return to save Wesnoth."
+            message= _ "I can hardly believe it! We didn’t need to destroy the bridge after all."
+        [/message]
+        [message]
+            speaker=Owaec
+            message= _ "Huzzah, a glorious victory in the name of the clans! Man will never flee from orc and drake; we have taught these northerners not to trifle with Wesnoth!"
+        [/message]
+        [message]
+            speaker=Dacyn
+            message= _ "(cough)(cough) Yes, yes, very well, well done. (cough) Now, it is finally time for us to see what has become of Wesnoth in our absence."
         [/message]
 
+        [set_achievement]
+            content_for=eastern_invasion
+            id=ei_S12
```

Comment:
I think it would be nice to use at least slightly more descriptive achievement names. Also, I'm pretty sure there's no need for an EI prefix, as achievements are namespaced within their achievement group. Though it's not like it hurts anything to use one anyway.